terminus-ui/drawer/src/drawer-container.component.ts

```diff
@@ -0,0 +1,368 @@
+// tslint:disable: template-no-call-expression
+import {
+  AfterContentInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ContentChild,
+  ContentChildren,
+  DoCheck,
+  ElementRef,
+  EventEmitter,
+  Inject,
+  Input,
+  NgZone,
+  OnDestroy,
+  Optional,
+  Output,
+  QueryList,
+  Renderer2,
+  ViewEncapsulation,
+} from '@angular/core';
+import { ANIMATION_MODULE_TYPE } from '@angular/platform-browser/animations';
+import { untilComponentDestroyed } from '@terminus/ngx-tools';
+import { TsUILibraryError } from '@terminus/ui/utilities';
+import { Subject } from 'rxjs';
+import {
+  debounceTime,
+  filter,
+  startWith,
+  take,
+  takeUntil,
+} from 'rxjs/operators';
+
+import { TsDrawerContentComponent } from './drawer-content.component';
+import { TsDrawerComponent } from './drawer.component';
+
+/**
+ * Throws an exception when two TsDrawer are matching the same position with same mode.
+ */
+export function throwTsDuplicatedDrawerError(position: string): void {
+  throw new TsUILibraryError(`TsDrawerContainer: Only one drawer can be in push mode on '"${position}" position'.`);
+}
+
+export interface ContentLeftRightMargin {
+  left: number | null;
+  right: number | null;
+}
+
+const MARGIN_SIZE_CONVERSION = 16;
```

Comment:
I know you mentioned this already, but since it's my last review I'm going to call them out 😬 

Missing comments